site/content/docs/main/restore-reference.md

```diff
@@ -36,57 +21,129 @@ Available Commands:
+1. When restoring with the [CSI plugin](csi.md), the PV is created from a CSI snapshot by the CSI driver. Velero doesn’t create the PV directly. Instead Velero creates a PVC with its DataSource referring to the CSI VolumeSnapshot object.
 
-- If the nodePort from the backup already allocated on the target cluster then Velero prints error log as shown below and continue to restore operation.
+### Snapshot PV Restore
 
-  ```
-  time="2020-11-23T12:58:31+03:00" level=info msg="Executing item action for services" logSource="pkg/restore/restore.go:1002" restore=velero/test-with-3-svc-20201123125825
+PV data backed up by durable snapshots is restored by VolumeSnapshot plugins. Velero calls the plugins’ interface to create a volume from a snapshot. The plugin returns the volume’s `volumeID`. This ID is created by storage vendors and will be updated in the PV object created by Velero, so that the PV object is connected to the volume restored from a snapshot.
 
-  time="2020-11-23T12:58:31+03:00" level=info msg="Restoring Services with original NodePort(s)" cmd=_output/bin/linux/amd64/velero logSource="pkg/restore/service_action.go:61" pluginName=velero restore=velero/test-with-3-svc-20201123125825
+### Restic PV Restore
 
-  time="2020-11-23T12:58:31+03:00" level=info msg="Attempting to restore Service: hello-service" logSource="pkg/restore/restore.go:1107" restore=velero/test-with-3-svc-20201123125825
+For more information on Restic restores, see the [Restic integration](restic.md#restore) page.
 
-  time="2020-11-23T12:58:31+03:00" level=error msg="error restoring hello-service: Service \"hello-service\" is invalid: spec.ports[0].nodePort: Invalid value: 31536: provided port is already allocated" logSource="pkg/restore/restore.go:1170" restore=velero/test-with-3-svc-20201123125825
-  ```
+### CSI PV Restore
 
+A PV backed up by CSI snapshots is restored by the [CSI plugin](csi). This happens when restoring the PVC object that has been snapshotted by CSI. The CSI VolumeSnapshot object name is specified with the PVC during backup as the annotation `velero.io/volume-snapshot-name`. After validating the VolumeSnapshot object, Velero updates the PVC by adding a `DataSource` field and setting its value to the VolumeSnapshot name.
 
+### Persistent Volume Rename
 
-- If the nodePort from the backup is not in the nodePort range of target cluster then Velero prints error log as below and continue to restore operation. Kubernetes default nodePort range is 30000-32767 but on the example cluster nodePort range is 20000-22767 and tried to restore Service with nodePort 31536
+Velero will rename a restored PV during restore if both of the following conditions are met:
 
-  ```
-  time="2020-11-23T13:09:17+03:00" level=info msg="Executing item action for services" logSource="pkg/restore/restore.go:1002" restore=velero/test-with-3-svc-20201123130915
+1. The PV already exists.
+1. The PV’s namespace has been [remapped](#restoring-into-a-different-namespace).
```

Comment:
Getting the meaning, but need to reword this part. PV is not a namespaced resource. The modified namespace should be PV's spec.claimRef.namespace.

Reply:
What about we call it "binding namespace" or "claiming namespace"? Because this the namespace of the PVC who is bound to it.

Reply:
goo call out @blackpiglet, updated the wording here. let me know if this works, thanks!